### ISSUE TITLE
Test the light client against the full node

### ISSUE THREAD

**Issue by tomaka** [smol-dot/smoldot#412]
The light client (everything under `light-base`) is currently poorly tested.
It should be possible to write tests against the full node.

- [ ] Split the full node into a library and binary.
- [ ] Add support for `/memory` multiaddresses in the full node library.
- [ ] Add a `light_base::Platform` trait implementation that allows connecting to `/memory` multiaddresses from the full node.
- [ ] Write the missing code in the full node, such as answering warp sync requests.
- [ ] Write the tests.

The tests in question would be about testing the high-level light client functionalities, such as whether connecting to a chain works at all. 
In other words, tests that could be written against, say, Westend.
The fact that we plug in the smoldot full node is a way to avoid having to be connected to the Internet to run tests, and not a way to have control over the behavior of the full node.
